### ISSUE TITLE
IMU reading drift and calibration

### ISSUE THREAD

**Issue by allenyin** [femtoduino/ArduinoCore-atsamd21e18a#12]
Hi Alex,

I've gone through all the steps calibrating the sensors with the FreeIMU-gui utility, saving it into the directory, etc.

I have also tried all sensor fusion algorithms (MARG option 0 to 4) and tweaked the settings in FreeIMU.h, under MPU9250_6511's parameters (see [my thread with FreeIMU-Updates](https://github.com/mjs513/FreeIMU-Updates/issues/80)).

I am wondering if you also see measurement drifts after calibration? If so, any tips on combating that?

Thanks,
Allen

**Comment by zrecore** [comment]
What version of the "Atmel SAM D21/R21 E" board package are you using?

**Comment by allenyin** [comment]
1.6.11

Are there anything in the core package that might affect the sensor fusion?

**Comment by zrecore** [comment]
Yes. See https://femtoduino.com/examples/setup

The latest MESH IMU Coin example uses ```getYawPitchRoll180()``` instead of ```getYawPitchRoll()```, and resets stuff on startup. I have ```#define MARG 4``` in FreeIMU.h, my local magnetic declination (```MAG_DEC```) and board defined as ```#define MPU9250_5611```

**Comment by allenyin** [comment]
Right I followed the documentation there for the setup.

I have changed the declination to my location. 

I am reading getEulers() since those are the actual orientation I want. According to the freeimu.cpp comments, the getyawpitchroll() method gives yaw as angle between earth north and IMU x-axis, rather than deviation from x-axis, if I'm understanding it correctly.

I will check again to see if using getyawpitchroll have any drift..
